metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py

```diff
@@ -772,10 +780,12 @@ def gen_dataset_workunits(
     def cache_tables_and_views(self, connection, database):
         tables, views = RedshiftDataDictionary.get_tables_and_views(conn=connection)
         for schema in tables:
-            if self.config.schema_pattern.allowed(f"{database}.{schema}"):
-                logging.debug(
-                    f"Not caching tables for schema {database}.{schema} which is not allowed by schema_pattern"
-                )
+            if is_schema_allowed(
+                self.config.schema_pattern,
+                schema,
+                database,
+                self.config.match_fully_qualified_names,
+            ):
```

Comment:
Can we keep this logic consistent with the above -- handle the `not` case first and `continue`? Same for above